executor/compiler.go

```diff
@@ -30,41 +30,76 @@ type Compiler struct {
 }
 
 func statementLabel(node ast.StmtNode) string {
-	switch node.(type) {
-	case *ast.SelectStmt:
-		return "Select"
-	case *ast.DeleteStmt:
-		return "Delete"
-	case *ast.InsertStmt:
-		if node.(*ast.InsertStmt).IsReplace {
-			return "Replace"
-		}
-		return "Insert"
-	case *ast.UnionStmt:
-		return "Union"
-	case *ast.UpdateStmt:
-		return "Update"
+	switch x := node.(type) {
+	case *ast.AlterTableStmt:
+		return "AlterTable"
+	case *ast.AnalyzeTableStmt:
+		return "AnalyzeTable"
+	case *ast.BeginStmt:
+		return "Begin"
+	case *ast.CommitStmt:
+		return "Commit"
+	case *ast.CreateDatabaseStmt:
+		return "CreateDatabase"
 	case *ast.CreateIndexStmt:
 		return "CreateIndex"
 	case *ast.CreateTableStmt:
 		return "CreateTable"
-	case *ast.CreateDatabaseStmt:
-		return "CreateDatabase"
+	case *ast.CreateUserStmt:
+		return "CreateUser"
+	case *ast.DeleteStmt:
+		return "Delete"
 	case *ast.DropDatabaseStmt:
 		return "DropDatabase"
-	case *ast.DropTableStmt:
-		return "DropTable"
 	case *ast.DropIndexStmt:
 		return "DropIndex"
+	case *ast.DropTableStmt:
+		return "DropTable"
+	case *ast.ExplainStmt:
+		return "Explain"
+	case *ast.InsertStmt:
+		if x.IsReplace {
+			return "Replace"
+		}
+		return "Insert"
+	case *ast.LoadDataStmt:
+		return "LoadData"
+	case *ast.RollbackStmt:
+		return "Rollback"
+	case *ast.SelectStmt:
+		if x.From == nil {
+			return "Select-Simple"
+		}
+		tableRefs := x.From.TableRefs
+		if tableRefs.Right == nil {
+			switch ref := tableRefs.Left.(type) {
+			case *ast.TableSource:
+				switch ref.Source.(type) {
+				case *ast.TableName:
+					return "Select-Simple"
+				}
+			}
+		}
+		return "Select-Complex"
+	case *ast.SetStmt, *ast.SetPwdStmt:
+		return "Set"
+	case *ast.ShowStmt:
+		return "Show"
+	case *ast.TruncateTableStmt:
+		return "TruncateTable"
+	case *ast.UpdateStmt:
+		return "Update"
+	case *ast.DeallocateStmt, *ast.ExecuteStmt, *ast.PrepareStmt, *ast.UseStmt:
+		return "Ignored"
 	}
-	return "unknown"
+	return "other"
 }
 
 // Compile compiles an ast.StmtNode to an ast.Statement.
 // After preprocessed and validated, it will be optimized to a plan,
 // then wrappped to an adapter *statement as stmt.Statement.
 func (c *Compiler) Compile(ctx context.Context, node ast.StmtNode) (ast.Statement, error) {
-	stmtNodeCounter.WithLabelValues(statementLabel(node)).Inc()
+	stmtCount(node)
```

Comment:
`stmtCount` is called in both `statement.Exec`  and here, would the metrics be calculated twice?